socialhome/streams/views.py

```diff
@@ -10,12 +12,67 @@ class BaseStreamView(ListView):
     ordering = "-created"
     paginate_by = 30
     stream_name = ""
+    vue = False
+    queryset = []
+
+    def dispatch(self, request, *args, **kwargs):
+        self.vue = bool(request.GET.get("vue", False))
+        return super().dispatch(request, *args, **kwargs)
 
     def get_context_data(self, **kwargs):
         context = super().get_context_data(**kwargs)
         context["stream_name"] = self.stream_name
+        if self.vue:
+            context["json_context"] = self._define_context_json(self.stream_name)
         return context
 
+    def _define_context_json(self, stream_name):
+        request_user_profile = getattr(self.request.user, "profile", None)
+
+        def dump_content(content):
+            is_user_author = (request_user_profile is not None and content.author == request_user_profile)
+            return {
+                "htmlSafe": content.rendered,
+                "author": {
+                    "guid": content.author.guid,
+                    "name": content.author.name if content.author.name else content.author.handle,
+                    "handle": content.author.handle,
+                    "imageUrlSmall": content.author.safer_image_url_small,
+                    "absoluteUrl": content.author.get_absolute_url(),
+                    "homeUrl": content.author.home_url,
+                    "isUserFollowingAuthor": (
+                        content.author.id in getattr(request_user_profile, "following_ids", []))
+                },
+                "id": content.id,
+                "timestamp": content.formatted_timestamp,
+                "humanizedTimestamp": content.humanized_timestamp,
+                "edited": bool(content.edited),
+                "childrenCount": content.children.count(),
+                "isUserLocal": bool(content.author.user),
+                "contentUrl": content.get_absolute_url(),
+                "isUserAuthor": is_user_author,
+                "updateUrl": reverse("content:update", kwargs={"pk": content.id}) if is_user_author else "",
+                "replyUrl": reverse("content:reply", kwargs={"pk": content.id}) if is_user_author else "",
+                "deleteUrl": reverse("content:reply", kwargs={"pk": content.id}) if is_user_author else ""
+            }
+
+        return {
+            "contentList": [dump_content(content) for content in self.queryset],
+            "currentBrowsingUserId": getattr(request_user_profile, "id", ""),
+            "streamName": stream_name,
+            "isUserAuthentificated": bool(self.request.user.is_authenticated),
+            "showAuthorBar": stream_name.startswith("profile_"),
```

Comment:
This should be `not stream_name.startswith("profile_")`? Unless I'm reading it wrong.

Basically author bar should be shown "if the stream is not a profile stream".